app/helpers/authentication_helper.rb

```diff
@@ -70,4 +70,8 @@ def tooltip_text_email_or_auth_provider_btns
   def came_from_sign_up?
     request.referer&.include?(new_user_registration_path)
   end
+
+  def display_social_login?
+    Authentication::Providers.enabled.include?(:apple) || request.user_agent.to_s.exclude?("ForemWebView")
```

Comment:
That's a great call! In the frontend we've checked for the Android platform also using the User Agent ([here](https://github.com/forem/forem/blob/main/app/assets/javascripts/initializers/runtime.js#L56)). I can add an extra check plus a spec to cover for this scenario here before merging 👍

Reply:
For context: We don't have any `ForemWebView` traffic from Android yet but I added the check to avoid this will interfere with displaying the providers when the time comes. On iOS the string `ForemWebView/<VERSION>` [is appended](https://github.com/forem/ForemWebView-ios/blob/main/Sources/ForemWebView/ForemWebView.swift#L56-L58) to the default browser's User Agent string, this helps keep the User Agent with an "proper structure" (like it were a normal browser but with a User Agent extension).

I added [this spec](https://github.com/forem/forem/pull/14260/files#diff-8311edf80693be929b85101291a3bd648db129a4590b3a67c529d7fc402ac658R139) to validate that it takes this case into account. For this I use a sample User Agent from a [Cypress test](https://github.com/forem/forem/blob/main/cypress/integration/loggedOutFlows/runtimeContext.spec.js#L39) we already have in place to differentiate the platforms.

Always ppen to hearing thoughts/opinions/questions on any of this though 🙂